pylops/signalprocessing/patch2d.py

```diff
@@ -271,5 +271,7 @@ def Patch2D(
         int(dims[0] // nwins0),
         int(dims[1] // nwins1),
     ), dimsd
+    OOp.dims = Pop.dims
```

Comment:
Earlier (combining0 * combining1 * OOp) would have been a LinearOperator but when we change `LinearOperator(Op=_ProductLinearOperator(self, Opx))` to `_ProductLinearOperator(self, Opx)`, it takes into consideration the dims of A and B and we run into an error in the `plot_patching` and `plot_sliding` 

`radon = Slid.H * data`
This command over here now that Slid would become a `_ProductLinearOperator`, its adjoint gave me those errors so I fixed it using there in sliding and patching files.

But Now as we are not making `Linear Operator ` the meta class we can write LinearOperator to instantiate it.
See errror in docs for [abf3f54](https://github.com/PyLops/pylops/pull/489/commits/abf3f54295ca1d400f31a481c1df0a016cea1752)

Reply:
Alright, let me take a look. What you did looks like a quick fix, I prefer going deep into the root of the problem and understand if it can be avoided within the LinearOperator class… I worry the same problem may now arise for other operators in users codes, so I’m not happy to have a quick fix unless it’s clear why it’s needed

Reply:
The other fix which seems to be correct is that we change the `_ProductLinearOperator(self, Opx)` to `LinearOperator(Op=_ProductLinearOperator(self, Opx))`

Reply:
I see the issue, some operators like `BlockDiag` do not propagate dims/dimsd as these don't make sense for such operators.

I think in this case we should just wrap `(combining0 * combining1 * OOp)` into a LinearOperator and assign dims/dimsd directly. I find this better than re-instating `LinearOperator` into `_ProductLinearOperator` as in most cases this is not needed. So now if we have `Op= Aop * Bop * Cop` we simply have two `_ProductLinearOperator`, whilst before (and if we bring back `LinearOperator` we will have twice wrap it into an extra `LinearOperator`

Reply:
Agreed this solves my query..